### PR TITLE
Add InitContainers and EphermalContainers to Require Limits and Requests

### PR DESCRIPTION
**Pull request by nikhilmaheshwari24**
Signed-off-by: Nikhil Maheshwari [nikhil.m2498@gmail.com](mailto:nikhil.m2498@gmail.com)

## Related Issue(s)
https://github.com/kyverno/policies/issues/951

## Description
This PR updates the existing Kyverno policy to validate CPU and memory resource requests and limits for both containers and initContainers within Pods. This enhancement ensures comprehensive resource management across all container types in a Kubernetes cluster.

## Checklist

- [x] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [x] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.
- [x] I have added the artifacthub-pkg.yml file and have verified it is complete and correct.

### CODE REVIEW

File: best-practices/require-pod-requests-limits/require-pod-requests-limits.yaml [hunk 1]
```diff
@@ -16,20 +16,34 @@ metadata:
       This policy validates that all containers have something specified for memory and CPU
       requests and memory limits.
 spec:
-  validationFailureAction: audit
+  validationFailureAction: Audit
   background: true
   rules:
-  - name: validate-resources
+  - name: validate-container-resources
```
Comment (chipzoller) [inline best-practices/require-pod-requests-limits/require-pod-requests-limits.yaml:22]: If you change this, CLI tests will probably break.

**Review by chipzoller — CHANGES_REQUESTED** [review]
Thanks for this. Couple things:
1. You must sign off on all commits (DCO).
2. See contribution guide [here](https://github.com/kyverno/policies?tab=readme-ov-file#contribution), specifically the part about updating Artifact Hub metadata file.

**Comment by realshuting** [comment]
Please sign-off your commits - https://github.com/kyverno/policies/pull/1103/checks?check_run_id=28177145454.

**Comment by nikhilmaheshwari24** [comment]
@realshuting @chipzoller Could you please approve this?

**Comment by chipzoller** [comment]
Unresolved conversations.

**Comment by chipzoller** [comment]
You have not followed the contribution guide, specifically the part about updating `artifacthub-pkg.yml` file.

**Review by chipzoller — APPROVED** [review]

**Comment by nikhilmaheshwari24** [comment]
@chipzoller I apologize. Have signed off the commits & updated the sha256sum to the `artifacthub-pkg.yml` file. 

I'll make sure such mistakes won't be repeated in future contributions.

**Comment by chipzoller** [comment]
No problem. If the guide isn't clear, let us know.

**Pull request merged**